lms/djangoapps/branding/__init__.py

```diff
@@ -17,14 +17,26 @@
 from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
 
 
-def get_visible_courses():
+def get_visible_courses(org=None):
     """
-    Return the set of CourseDescriptors that should be visible in this branded instance
+    Return the set of CourseOverviews that should be visible in this branded instance
     """
-    filtered_by_org = microsite.get_value('course_org_filter')
-    courses = CourseOverview.get_all_courses(org=filtered_by_org)
+    microsite_org = microsite.get_value('course_org_filter')
+
+    if org and microsite_org:
+        # When called in the context of a microsite, return an empty result if the org
+        # passed by the caller does not match the designated microsite org.
+        courses = CourseOverview.get_all_courses(org=org) if org == microsite_org else []
+    else:
+        target_org = org or microsite_org
+        courses = CourseOverview.get_all_courses(org=target_org)
```

Comment:
Thanks for the look @chrisndodge. The approach you're describing is actually what I'd implemented prior to this. I changed it to the implementation proposed here in the interest of performance. I'll add my note to @nasthagiri below as a comment here to help others understand this more quickly.